### ISSUE TITLE
Optional _hot sorting_ of conversation stream

### ISSUE THREAD

**Issue by cpt1gl0** [diaspora/diaspora#6843]
Add a switch to the top of the conversation stream which allows to toggle between _new sorting_ and _hot sorting_.

_New sorting_ corresponds to the usual sorting of conversations by creation time.
_Hot sorting_ relies on a ranking algorithm which takes into account creation time, but also the number of likes (upvotes) for each individual post.

For _hot sorting_ the ranking algorithm from reddit is used. The implementation is simple and can be taken from reddits source code:
https://github.com/reddit/reddit/blob/master/r2/r2/lib/db/_sorts.pyx

The corresponding Loomio discussion can be found here:
https://www.loomio.org/d/fudqSbQC/hot-post-sorting

**Comment by cpt1gl0** [comment]
Is someone going to implement this?

**Comment by svbergerem** [comment]
@cpt1gl0 There are no likes for conversations. Which conversation stream are you talking about? The “My activity” stream? These posts aren't sorted by creation date but by the date someone interacted with the post. (e.g. liked the post, etc.)

**Comment by cpt1gl0** [comment]
@svbergerem What I actually meant was the _Stream_ and not the conversation stream. I am sorry for the confusion.

Ideally the sorting switch should be available for all streams: The _Stream_ stream, the _My Activity_ stream, the @Mentions stream, etc. Depending on the respective stream an appropriate sorting should be preselected. As I see it (please correct me if I am wrong) at the moment there is only a sorting by creation time and another sorting by last activity in the case of the _My Activity_ stream. So for all streams there should be a switch to switch between:
 - Sorting by creation time
 - Sorting by last activity
 - Hot sorting (the new one)

**Comment by Flaburgan** [comment]
I had a similar idea a few months ago but with a major difference, it would not be oriented to *your* content like here, but to the network content. So it would not be a way of sorting your streams, but a new stream containing all the posts you have access to (private ones + public ones) ordered with that algorithm. I called it the "Popular" stream. The algorithm is a formula with the date of the last interaction on the post, the number of likes, reshares and comments. I unfortunately lose my draft on a paper with my maths, but I can probably do it again ;) This is the kind of stream which could be displayed as suggested in https://github.com/diaspora/diaspora/issues/3618

**Comment by cpt1gl0** [comment]
Yes, maybe we should go with a "Popular" stream instead. In that way we do not need to extend the existing GUI in any way, except for the additional link to the "Popular" stream.

Concerning the algorithm I would prefer to use the Reddit hot ranking algorithm and not to cook up something new. After all the Reddit algorithm is already out in the wild for quite some time and seems to do what it is expected to do.

If we should go with the Reddit algorithm I'd prefer to call the stream "Hot".

**Comment by Flaburgan** [comment]
Fyi I started working on this some weeks ago. Hard to find time to do it but I'm going to assign myself anyway, and hopefully this will become real before the summer.
